backtesting/_stats.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pandas as pd
+
+from ._util import _data_period
+
+
+def compute_drawdown_duration_peaks(dd: pd.Series):
+    iloc = np.unique(np.r_[(dd == 0).values.nonzero()[0], len(dd) - 1])
+    iloc = pd.Series(iloc, index=dd.index[iloc])
+    df = iloc.to_frame('iloc').assign(prev=iloc.shift())
+    df = df[df['iloc'] > df['prev'] + 1].astype(int)
+    # If no drawdown since no trade, avoid below for pandas sake and return nan series
+    if not len(df):
+        return (dd.replace(0, np.nan),) * 2
+    df['duration'] = df['iloc'].map(dd.index.__getitem__) - df['prev'].map(dd.index.__getitem__)
+    df['peak_dd'] = df.apply(lambda row: dd.iloc[row['prev']:row['iloc'] + 1].max(), axis=1)
+    df = df.reindex(dd.index)
+    return df['duration'], df['peak_dd']
+
+
+def compute_stats(data, broker, strategy) -> pd.Series:
```

Comment:
In order to expose this function as API (or, rather, an additional wrapper of this function in `backtesting.lib`), we need to revise its inputs since `broker` is not for public to use. I think somewhat like [proposed](https://github.com/kernc/backtesting.py/issues/194#issuecomment-743762084), with `trades=` and `equity=` from [trade data](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Trade-data).